### PR TITLE
Return nil instead of NIL for vim.env

### PR DESCRIPTION
**Pull request by norcalli**

### CODE REVIEW

File: test/functional/lua/vim_spec.lua [hunk 1]
```diff
@@ -569,7 +569,7 @@ describe('lua stdlib', function()
     vim.fn.setenv("A", 123)
     ]]
     eq('123', funcs.luaeval "vim.env.A")
-    eq(NIL, funcs.luaeval "vim.env.B")
+    eq(nil, funcs.luaeval "vim.env.B")
```
Comment (bfredl) [inline test/functional/lua/vim_spec.lua]: both of these `luaeval` to `v:null`. You need to use `vim.env.B == nil` instead

Comment (justinmk) [reply test/functional/lua/vim_spec.lua]: or check for `mpack.NIL`

```suggestion
    eq(mpack.NIL, funcs.luaeval "vim.env.B")
```

Comment (bfredl) [reply test/functional/lua/vim_spec.lua]: That is just the existing code.

Comment (norcalli) [reply test/functional/lua/vim_spec.lua]: What's werid is that my local tests are passing fine.

**Pull request merged**